filter/pdftopdf/pdftopdf.cc

```diff
@@ -750,6 +771,15 @@ bool checkFeature(const char *feature, int num_options, cups_option_t *options)
 	      (param.page_logging == 0 ? "not " : ""));
     }
   }
+  /*
+   *  To make sure fillprint or cropfit is not applied twice.
+   */
+  char* content_type = getenv("CONTENT_TYPE");
+  if(!strncasecmp(content_type,"image",5))
+  {
+    param.fillprint =0;
+    param.cropfit =0;
+  }
```

Comment:
@tillkamppeter Is this check sufficient to make sure fillprint and cropfit are not applied again in ```pdftopdf``` filter?

Reply:
You are checking via
```
char* content_type = getenv("CONTENT_TYPE");
￼if(!strncasecmp(content_type,"image",5)) ...
```
meaning that if the file type in which the job got sent is any MIME type starting with "image". This would also include "image/urf", "image/pwg-raster", ... which are formats not treated by the "`imageto...`" filters and so they need the "fit-to-page", "fill", "crop-to-fit" option be applied by "`pdftopdf`".
If it comes to the use of "`imagetoraster`", "`pdftopdf`" is never called. If "`imagetopdf`" is used we must decide. If we use "`imagetopdf`" we need to make it mark its PDF output, so that "`pdftopdf`" knows that there was "`imagetopdf`" before and "fit-to-page", "fill", "crop-to-fit" do not need to be done any more. The marking can be done the same way as "`pdftopdf`" does to tell "`pdftops`" about Copies and Collate, by adding a comment. "`imagetopdf`" could add "%%from imagetopdf" and "`pdftopdf`" checks whether the input stream contains it, then does not apply the scaling/crop options any more.
Please try it this way and test.

Reply:
Hi Till,
I noticed that imagetopdf gives ```application/vnd.cups-pdf``` as output format. 
```
 ~/g/c/mime> grep -R "imagetopdf" *
cupsfilters.convs:image/gif		application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs:image/png		application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs:image/jpeg		application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs:image/tiff		application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs:image/x-bitmap		application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs:image/x-photocd		application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs:image/x-portable-anymap	application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs:image/x-portable-bitmap	application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs:image/x-portable-graymap application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs:image/x-portable-pixmap	application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs:image/x-sgi-rgb		application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs:image/x-xbitmap		application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs:image/x-xpixmap		application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs:image/x-xwindowdump	application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs:image/x-sun-raster	application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs.in:image/gif		application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs.in:image/png		application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs.in:image/jpeg		application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs.in:image/tiff		application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs.in:image/x-bitmap		application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs.in:image/x-photocd		application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs.in:image/x-portable-anymap	application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs.in:image/x-portable-bitmap	application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs.in:image/x-portable-graymap application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs.in:image/x-portable-pixmap	application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs.in:image/x-sgi-rgb		application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs.in:image/x-xbitmap		application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs.in:image/x-xpixmap		application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs.in:image/x-xwindowdump	application/vnd.cups-pdf		65	imagetopdf
cupsfilters.convs.in:image/x-sun-raster	application/vnd.cups-pdf		65	imagetopdf

```
and pdftopdf converts ```application/pdf``` to ```application/vnd.cups-pdf```: 
```
 ~/g/c/mime> grep -R "pdftopdf" *
cupsfilters.convs:# that pdftopdf is doing the page management on PDF data and the
cupsfilters.convs:application/pdf		application/vnd.cups-pdf		66	pdftopdf
cupsfilters.convs.in:# that pdftopdf is doing the page management on PDF data and the
cupsfilters.convs.in:application/pdf		application/vnd.cups-pdf		66	pdftopdf
```
So, imagetopdf and pdftopdf will not work on same input. Am I missing something?

Reply:
OK, you are right. Then this problem will not occur.
Only make sure that when it comes to an `imagetopdf`-based filter chain that the common page management options (if they apply, does `imagetopdf` accept any multi-page format as input ?) work.

Reply:
Which functionalities do I need to test? I don't think ```imagetopdf```  accepts multi-page format. One known issue is, If we specify orientation with ```crop-to-fit``` we get wrong results in ```imagetopdf``` and ```imagetoraster```. ```crop-to-fit``` is not intended to be used with orientation. Is there any way to ignore orientation if ```crop-to-fit``` is specified?

Reply:
I see that multi-page output is only possible if you scale the image larger than the output page size to allow for a multi-page poster. With this duplex, reverse order, collate, number-up, and selected pages make sense, but applying any option which you have added your force the filter to take a single input page and produce a single output page from it. So page management options do not make sense.
crop-to-fit does only make sense for input which has an absolute size defined, like PDF or PostScript. On images, like photos, which do not have an absolute size defined (only being a bitmap) crop-to-fit does not make sense and should be ignored (fall back to default, fit-to-page).
Orientation with crop-to-fit is handled as without crop-to-fit. If not forced by orientation-requested the page is rotated if its orientation does not match the output page orientation (portrait on most printers) and after that crop-to-fit is applied.
So please make sure that it behaves this way and if so, tell me, so that I can merge the pull request.

Reply:
I have made the required changes. PR is ready from my side. Please go through the PR once more.